src/cluster_asm.c

```diff
@@ -1985,10 +1986,17 @@ void clusterSyncSlotsCommand(client *c) {
                                 "updated dest offset to %lld, source offset: %lld",
                 asmTaskStateToString(dest_state), task->dest_offset, task->source_offset);
 
+            /* Record the time when the destination finishes applying the accumulated buffer */
+            if (task->dest_state == ASM_WAIT_STREAM_EOF && task->dest_accum_applied_time == 0)
+                task->dest_accum_applied_time = server.mstime;
+
```

Comment:
```
*** [err]: Sync buffer drain timeout in tests/unit/cluster/atomic-slot-migration.tcl
ASM task did not enter wait-stream-eof state
```

Is this related to these test failures or not?

Reply:
yep, even i set lag size to 0, it still happened, so i use fail point to guarantee we do not enter handoff prep state.

as below, we return if fail point active, so we should update time here